File: .tmp_py4013/batch_08_dump/2903_find_indices_with_index_and_value_difference_i.js
// LeetCode 2903 - Find Indices With Index and Value Difference I
// https://leetcode.com/problems/find-indices-with-index-and-value-difference-i/

/**
 * @param {number[]} nums
 * @param {number} indexDifference
 * @param {number} valueDifference
 * @return {number[]}
 */
var findIndices = function(nums, indexDifference, valueDifference) {
    const n = nums.length;
    for (let i = 0; i < n; i++)
        for (let j = i; j < n; j++) {
            if (Math.abs(j - i) >= indexDifference && Math.abs(nums[i] - nums[j]) >= valueDifference)
                return [i, j];
        }
    return [-1, -1];
};

File: .tmp_rs4013/rust_check_12/2903_find_indices_with_index_and_value_difference_i.rs
struct Solution;
// LeetCode 2903 - Find Indices With Index and Value Difference I
// https://leetcode.com/problems/find-indices-with-index-and-value-difference-i/

impl Solution {
    pub fn find_indices(nums: Vec<i32>, index_difference: i32, value_difference: i32) -> Vec<i32> {
        let n = nums.len();
        for i in 0..n {
            for j in i..n {
                let di = (j as i32 - i as i32).abs();
                let dv = (nums[i] - nums[j]).abs();
                if di >= index_difference && dv >= value_difference {
                    return vec![i as i32, j as i32];
                }
            }
        }
        vec![-1, -1]
    }
}

fn main() {}
